features/ssl/README.md

```diff
@@ -0,0 +1,180 @@
+keytool -import -v -trustcacerts -alias <cassandra_node1> -file cassandranode1.cer -keystore client.truststore
+...
+```
+
+If you're using a Certificate Authority, the client truststore only
+needs to contain the CA's certificate:
+
+```
+keytool -import -v -trustcacerts -alias CARoot -file ca.cer -keystore client.truststore
+```
+
+#### Client keystore
+
+If you also intend to use client certificate authentication, generate
+the public and private key pair for the client:
+
+```
+keytool -genkey -keyalg RSA -alias client -keystore client.keystore
+```
+
+If you're using self-signed certificates, extract the public part of the
+client certificate, and import it in the truststore of each Cassandra
+node:
+
+```
+keytool -export -alias client -file client.cer -keystore client.keystore
+keytool -import -v -trustcacerts -alias client -file client.cer -keystore server.truststore
+```
+
+If you're using a CA, sign the client certificate with it (see the blog
+post linked at the top of this page). Then the nodes' truststores only
+need to contain the CA's certificate (which should already be the case
+if you've followed the steps for inter-node encryption).
+
+### Driver configuration
+
+The base class to configure SSL is [SSLOptions]. It's very generic, but
+you don't necessarily need to deal with it directly: the default
+instance, or the provided subclasses, might be enough for your needs.
+
+#### JSSE, Property-based
+
+`withSSL()` gives you a basic JSSE configuration:
+
+```java
+Cluster cluster = Cluster.builder()
+  .addContactPoint("127.0.0.1")
+  .withSSL()
+  .build();
+```
+
+You can then use
+[JSSE system properties](http://docs.oracle.com/javase/6/docs/technotes/guides/security/jsse/JSSERefGuide.html#Customization)
+for specific details, like keystore locations and passwords:
+
+```
+-Djavax.net.ssl.trustStore=/path/to/client.truststore
+-Djavax.net.ssl.trustStorePassword=password123
+# If you're using client authentication:
+-Djavax.net.ssl.keyStore=/path/to/client.keystore
+-Djavax.net.ssl.keyStorePassword=password123
+```
+
+#### JSSE, programmatic
+
+If you need more control than what system properties allow, you can
+configure SSL programmatically with [JdkSSLOptions]:
+
+```java
+SSLContext sslContext = ... // create and configure SSL context
+
+JdkSSLOptions sslOptions = JdkSSLOptions.builder()
+  .withSSLContext(context)
+  .build();
+
+Cluster cluster = Cluster.builder()
+  .addContactPoint("127.0.0.1")
+  .withSSL(sslOptions)
+  .build();
+```
+
+Note that you can also extend the class and override `newSSLEngine()` if
+you need specific configuration on the `SSLEngine` (for example hostname
+verification).
+
+#### Netty
+
+[NettySSLOptions] allows you to use Netty's `SslContext` instead of
+the JDK directly. The advantage is that Netty can use OpenSSL if
+available, which provides better performance.
+
+##### Converting your client certificates for OpenSSL
```

Comment:
This section is a bit out of scope from the driver's perspective imho, but it's short, so let's leave it.